x-pack/plugins/observability_onboarding/public/components/app/custom_logs/install_elastic_agent.tsx

```diff
@@ -113,7 +113,7 @@ export function InstallElasticAgent() {
         return callApi('POST /internal/observability_onboarding/logs/flow', {
           params: {
             body: {
-              name: datasetName,
+              name: datasetName ?? '',
```

Comment:
It feels like a code smell to default to an empty string here. If `datasetName` is required we should make the call conditional like:

```ts
if(datasetName) {
  return callApi('POST /internal/observability_onboarding/logs/flow', {
  // ...
  })
}
```